dts/msm8916/msm8916-512mb-mtp.dts

```diff
@@ -17,5 +17,22 @@
 		/* Register EDL button as HOME button */
 		/* This is the only button on this device */
 		lk2nd,keys = <KEY_HOME 37 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
+		lk2nd,match-cmdline = "* mdss_mdp.panel=1:spi:0:qcom,mdss_spi_st7735s_128128_cmd";
+	};
+
+	/*
+	 * Note:
+	 * Due to some unknown reasons, the stock aboot firmware is incompatible
+	 * with qhypstub and tz firmware from db410c, use lk1st if possible.
+	 */
```

Comment:
Is it really worth supporting lk2nd on those at all? They'd probably be much better suited with lk1st.

Reply:
> Is it really worth supporting lk2nd on those at all? They'd probably be much better suited with lk1st.

I don't really want to support lk2nd. lk1st is confirmed to work quite well on almost every msm8916-based dongles. On the contrary, lk2nd is often broken due to quirky stock bootloader provided by vendors. Even sbl1 can be ported from db410c to avoid unnecessary problems. However, since the project is named after "lk2nd", is it acceptable to drop the support of lk2nd?

Reply:
It would be nice if you could make sure that booting lk2nd from lk1st (rather than the stock bootloader) works. This should still allow dropping most of the necessary quirks.

Reply:
> It would be nice if you could make sure that booting lk2nd from lk1st (rather than the stock bootloader) works. This should still allow dropping most of the necessary quirks.

So qhypstub + tz from db410c + lk1st + lk2nd? Probably works. I remembered testing it before.

> It would be nice if you could make sure that booting lk2nd from lk1st (rather than the stock bootloader) works. This should still allow dropping most of the necessary quirks.

> It would be nice if you could make sure that booting lk2nd from lk1st (rather than the stock bootloader) works. This should still allow dropping most of the necessary quirks.

So qhypstub + tz from db410c + lk1st + lk2nd? Probably works. I remembered testing it before.

Reply:
Yes, exactly.

Reply:
> Hello! I can confirm that lk2nd can be booted from lk1st on my uz801 with db410c firmware.
> 
> ```
> $ fastboot getvar product
> product: LK2ND_MSM8916
> ```

I've confirmed again mine is not able to work with qhypstub+tz firmware from db410c.
To ensure we are talking about the same tz firmware, please look up the size of your tz firmware, it should be around ~400 KB, rather than the tz firmware from openstick, which is about 600KB.

Reply:
> Hello! I can confirm that lk2nd can be booted from lk1st on my uz801 with db410c firmware.
> 
> ```
> $ fastboot getvar product
> product: LK2ND_MSM8916
> ```

Please try again with this link: https://releases.linaro.org/96boards/dragonboard410c/linaro/rescue/17.09/dragonboard410c_bootloader_emmc_android-88.zip
The tz firmware from here is smaller than 512kb, can be flashed directly to tz partition, and should be the suggested tz firmware rather than the larger one you may encounter elsewhere.

Reply:
> Please try again with this link: https://releases.linaro.org/96boards/dragonboard410c/linaro/rescue/17.09/dragonboard410c_bootloader_emmc_android-88.zip The tz firmware extracted from it is smaller than 512kb, can be flashed directly to tz partition, and should be the suggested tz firmware rather than the larger one you may encounter elsewhere.

Ah, my mistake, I'm replying to stephan-gh, the lk1st here is not the stock aboot, it is compiled from your branch

And this is how I flashed the firmware (without the modifications to partition.txt): https://github.com/iovxw/device-thwc-ufi001c/tree/master/partitions